### PR TITLE
Add default Retry-After value

### PR DESCRIPTION
**Pull request by mattjohnsonpint**
Completes #1443 

Refactored the logic for better readability as well.

### CODE REVIEW

File: src/Sentry/Internal/Http/RetryAfterHandler.cs [hunk 1]
```diff
@@ -19,6 +19,7 @@ internal class RetryAfterHandler : DelegatingHandler
         private readonly ISystemClock _clock;
 
         private const HttpStatusCode TooManyRequests = (HttpStatusCode)429;
+        internal static readonly TimeSpan DefaultRetryAfterDelay = TimeSpan.FromSeconds(60);
```
Comment (SimonCropp) [inline src/Sentry/Internal/Http/RetryAfterHandler.cs:22]: `TimeSpan.FromMinutes(1)`

File: src/Sentry/Internal/Http/RetryAfterHandler.cs [hunk 2]
```diff
@@ -63,31 +64,43 @@ internal RetryAfterHandler(HttpMessageHandler innerHandler, ISystemClock clock)
...
+                var retryAfterTimestamp = GetRetryAfterTimestamp(response);
+                _ = Interlocked.Exchange(ref _retryAfterUtcTicks, retryAfterTimestamp.UtcTicks);
+            }
+
+            return response;
+        }
+
+        private DateTimeOffset GetRetryAfterTimestamp(HttpResponseMessage response)
+        {
+            if (response.Headers != null)
```
Comment (SimonCropp) [inline src/Sentry/Internal/Http/RetryAfterHandler.cs:78]: the code for header is 
```
public HttpResponseHeaders Headers => _headers ??= new HttpResponseHeaders();
```
so the null check would seem redundant?

File: src/Sentry/Internal/Http/RetryAfterHandler.cs [hunk 3]
```diff
@@ -63,31 +64,43 @@ internal RetryAfterHandler(HttpMessageHandler innerHandler, ISystemClock clock)
...
-                        _ = Interlocked.Exchange(ref _retryAfterUtcTicks, response.Headers.RetryAfter.Date.Value.UtcTicks);
+                        return date;
                     }
                 }
+
                 // Sentry was sending floating point numbers which are not handled by RetryConditionHeaderValue
                 // To be compatible with older versions of sentry on premise: https://github.com/getsentry/sentry/issues/7919
-                else if (response.Headers.TryGetValues("Retry-After", out var values)
-                         && double.TryParse(values?.FirstOrDefault(), out var retryAfterSeconds))
+                if (response.Headers.TryGetValues("Retry-After", out var values)
```
Comment (SimonCropp) [inline src/Sentry/Internal/Http/RetryAfterHandler.cs:95]: TryGetValues has a notnullwhentrue for value. so the following `values?` is redundant?

File: src/Sentry/Internal/Http/RetryAfterHandler.cs (hunk 1 above)
Comment (mattjohnsonpint) [reply src/Sentry/Internal/Http/RetryAfterHandler.cs:22]: Thanks, but I'd rather keep it in terms of seconds, to match the spec.

**Review by SimonCropp — APPROVED** [review]

**Pull request merged**